### ISSUE TITLE
Case splitting prints out-of-scope pattern synonyms

### ISSUE THREAD

**Issue by andreasabel** [agda/agda#3080]
```agda
module Issue3080 where

open import Agda.Builtin.Nat
open import Agda.Builtin.Equality

data Fin (m : Nat) : Set where
  fzero' : (n : Nat) (p : m ≡ suc n) → Fin m
  fsuc'  : (n : Nat) (p : m ≡ suc n) (i : Fin n) → Fin m

lift : (m : Nat) (i : Fin m) → Fin (suc m)
lift m (fzero' n p)  = {!p!}
lift m (fsuc' n p i) = {!p!} -- Split on p here

module PatternSynonyms where

  pattern fzero n      = fzero' n refl
  pattern fsuc  n i    = fsuc'  n refl i

-- Splitting produces
--
--   lift .(suc n) (.Issue3080.PatternSynonyms.fsuc n i) = ?
--
-- which fails to parse.
```
Pattern synonyms which are not in scope should not be used for printing.

Works fine in Agda 2.5.3.

**Comment by andreasabel** [comment]
See also the more general open discussion at #2902.
Maybe we should exclude feature #2762 from the 2.5.4 release.

**Comment by andreasabel** [comment]
Fixed by @UlfNorell by checking whether pattern synonym is in scope before reconstructing it.

**Comment by UlfNorell** [comment]
> Fixed by @UlfNorell by checking whether pattern synonym is in scope before reconstructing it.

Um, yes... See above:

>  UlfNorell closed this in be86cdf 11 hours ago
